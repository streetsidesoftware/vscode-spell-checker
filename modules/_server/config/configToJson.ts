import type { CSpellSettings } from '@cspell/cspell-types';

export function configToJson(config: CSpellSettings, exclude: string[] = []): string {
    function replacer(_key: string, value: unknown) {
        if (value instanceof RegExp) {
            return value.toString();
        }
        return value;
    }

    const obj: Record<string, any> = { ...config };
    exclude.forEach((key) => delete obj[key]);

    return JSON.stringify(obj, replacer, 2);
}
