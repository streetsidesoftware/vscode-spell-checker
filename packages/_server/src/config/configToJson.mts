import type { CSpellSettings } from '@cspell/cspell-types';

export function configToJson(config: CSpellSettings, exclude: string[] = []): string {
    function replacer(_key: string, value: unknown) {
        if (value instanceof RegExp) {
            return value.toString();
        }
        return value;
    }

    const excludeSet = new Set(exclude);

    const obj: Record<string, unknown> = Object.fromEntries(Object.entries(config).filter(([key]) => !excludeSet.has(key)));

    return JSON.stringify(obj, replacer, 2);
}
