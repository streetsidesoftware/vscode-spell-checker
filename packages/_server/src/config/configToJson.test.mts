import type { CSpellSettingsWithSourceTrace } from 'cspell-lib';
import { describe, expect, test } from 'vitest';

import { configToJson } from './configToJson.mjs';

describe('Validate configToJson', () => {
    const sampleConfig: CSpellSettingsWithSourceTrace = {
        version: '0.2',
        patterns: [
            {
                name: 'word',
                pattern: /^\s*word/gim,
            },
        ],
        ignoreRegExpList: [/'.*'/g],
        source: {
            name: 'source',
        },
    };

    test.each`
        cfg             | message
        ${{}}           | ${'empty config'}
        ${sampleConfig} | ${'sampleConfig'}
    `('configToJson $message', ({ cfg }) => {
        expect(configToJson(cfg, ['source'])).toMatchSnapshot();
    });
});
