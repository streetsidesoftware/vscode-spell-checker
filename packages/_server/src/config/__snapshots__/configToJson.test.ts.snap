// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Validate configToJson > configToJson 'empty config' 1`] = `"{}"`;

exports[`Validate configToJson > configToJson 'sampleConfig' 1`] = `
"{
  \\"version\\": \\"0.2\\",
  \\"patterns\\": [
    {
      \\"name\\": \\"word\\",
      \\"pattern\\": \\"/^\\\\\\\\s*word/gim\\"
    }
  ],
  \\"ignoreRegExpList\\": [
    \\"/'.*'/g\\"
  ]
}"
`;
